flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraph.java

```diff
@@ -97,6 +97,8 @@
 
 	private TimeCharacteristic timeCharacteristic;
 
+	private boolean blockingAfterChainingOff;
```

Comment:
I think `slotSharingEnableAfterChain` is more accurate?  From the discussion in the slack what my understand is that the planner only controls whether slot is sharing and ShuffleMode is determined by this currently.
What do you think?

Reply:
I'm not sure we have reached an agreement of details. I guess we have a divergence here.
Let's wait for the opinion of Aljoscha and Stephan.

Reply:
ok. I just remind another thing that Blink Planner might set "slot-sharing after chaining" off and ask ShuffleMode between tasks to be PIPELINE.